packages/sources/accuweather/src/endpoint/current-conditions.ts

```diff
@@ -0,0 +1,262 @@
+    currentConditionsResult.precipitationPastHour * 100,
+  )
+  currentConditionsResult.pressure = Math.trunc(currentConditionsResult.pressure * 100)
+  currentConditionsResult.temperature = Math.trunc(currentConditionsResult.temperature * 10)
+  currentConditionsResult.windSpeed = Math.trunc(currentConditionsResult.windSpeed * 10)
+
+  // NB: currently all conditions are expected to be numbers
+  Object.entries(currentConditionsResult).forEach(([key, value]) => {
+    if (Number.isNaN(value))
+      throwError(
+        `Attribute '${key}' is not a number: ${JSON.stringify(
+          currentConditionsResult,
+        )}. Current conditions are: ${JSON.stringify(currentConditions)}`,
+      )
+  })
+
+  return currentConditionsResult
+}
+
+export const encodeCurrentConditionsResult = (result: CurrentConditionsResult): string => {
+  const dataTypes = [
+    'tuple(uint256,uint24,uint24,uint24,uint24,int16,uint16,uint16,uint8,uint8,uint8,uint8)',
+  ]
+  const dataValues = [
+    [
+      result.timestamp, // uint256
+      result.precipitationPast12Hours, // uint24
+      result.precipitationPast24Hours, // uint24
+      result.precipitationPastHour, // uint24
+      result.pressure, // uint24
+      result.temperature, // int16
+      result.windDirectionDegrees, // uint16
+      result.windSpeed, // uint16
+      result.precipitationType, // uint8
+      result.relativeHumidity, // uint8
+      result.uvIndex, // uint8
+      result.weatherIcon, // uint8
+    ],
+  ]
+
+  return utils.defaultAbiCoder.encode(dataTypes, dataValues)
+}
+
+export const execute: ExecuteWithConfig<Config> = async (request, _, config) => {
+  const validator = new Validator(request, inputParameters)
+  if (validator.error) throw validator.error
+
+  const jobRunID = validator.validated.id
+  const locationKey = validator.validated.data.locationKey
+  const units = validator.validated.data.units
+  const encodeResult = validator.validated.data.encodeResult ?? true
+
+  validateUnitsParameter(jobRunID, units)
+
+  const url = `currentconditions/v1/${locationKey}.json`
+  const params = {
+    details: true,
+    apikey: config.apiKey,
+  }
+
+  const options = { ...config.api, params, url }
+
+  const response = await Requester.request<Array<CurrentConditions>>(options)
+
+  const currentConditionsList = response.data
+  if (!Array.isArray(currentConditionsList)) {
+    throw new AdapterError({
+      jobRunID,
+      message: `Unexpected response by location key: ${locationKey}. Expected an array but got: ${JSON.stringify(
+        currentConditionsList,
+      )}.`,
+      statusCode: 400,
+    })
+  }
+  let currentConditionsResult: CurrentConditionsResult
+  try {
+    currentConditionsResult = getCurrentConditionsResult(units, currentConditionsList)
+  } catch (error) {
+    throw new AdapterError({
+      jobRunID,
+      message: `Unprocessable response by location key: ${locationKey}. ${error}.`,
+      statusCode: 400,
+    })
```

Comment:
We shouldn't need to throw `AdapterError` here.
We'll catch it later and add relevant context, so we should just throw a standard `Error` here

Reply:
Sorry, by catching it later you mean in the `location-current-conditions` endpoint?

Due to `location` and `current-conditions` are exposed endpoints too, I throw `AdapterError` in each one.
 
I get that the `location` endpoint is not very "appealing" and probably no one will ever set a job for it. On the contrary, I see value/utility on the `current-conditions` once you know the location key.

What do you think?

Reply:
So we can just throw a regular error here:
```suggestion
    throw new Error(`Unprocessable response by location key: ${locationKey}. ${error}.`)
```

The [EA framework will catch this](https://github.com/smartcontractkit/external-adapters-js/blob/develop/packages/core/bootstrap/src/index.ts#L213-L227) and give the appropriate feed ID, job run ID, etc. The only thing you gain from the current implementation is that you can set an appropriate status code.

Reply:
done!